config/awssqs/istio_egress.yaml

```diff
@@ -0,0 +1,14 @@
+# Needed by the AWS SQS Source to communicate with AWS SQS.
+
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: awssqs-bus-ext
+spec:
+  hosts:
+  - "*.amazonaws.com"
```

Comment:
This is actually not working and I had to use the FQDN (sqs.eu-west-1.amazonaws.com in my case) to make it work.

Reply:
Thanks @rootfs. It looks like a false alarm.  The istio-proxy container takes longer to start than the receive-adapter so you can see this in the logs:

> default/awssqs-devhose-source-vzrsc-765c547b68-gmp7m[receive-adapter]: {"level":"info","ts":1542735771.9919236,"caller":"awssqs_receive_adapter/main.go:68","msg":"Starting AWS SQS Receive Adapter. %v","adapter":{"Region":"eu-west-1","QueueUrl":"...","SinkURI":"http://qux-1-channel.default.svc.cluster.local/","CredsFile":"/var/secrets/aws/credentials"}}
> default/awssqs-test-source-vzrsc-765c547b68-gmp7m[receive-adapter]: {"level":"info","ts":1542735771.9922512,"logger":"fallback","caller":"awssqs/adapter.go:63","msg":"Starting with config: {adapter 22 0  0xc4200cedc0}"}
> default/awssqs-test-source-vzrsc-765c547b68-gmp7m[receive-adapter]: {"level":"warn","ts":1542735772.3487105,"logger":"fallback","caller":"awssqs/adapter.go:77","msg":"Failed to poll from SQS queue{error 25 0  RequestError: send request failed\ncaused by: Post https://sqs.eu-west-1.amazonaws.com/: dial tcp 52.95.114.128:443: connect: connection refused}"}

But the istio-proxy does initialize correctly after a while

> default/awssqs-devhose-source-vzrsc-765c547b68-gmp7m[istio-proxy]: [2018-11-20 17:42:53.579][19][info][upstream] external/envoy/source/common/upstream/cluster_manager_impl.cc:494] add/update cluster outbound|80||message-dumper-00001-service.default.svc.cluster.local during init
> default/awssqs-devhose-source-vzrsc-765c547b68-gmp7m[istio-proxy]: [2018-11-20 17:42:53.579][19][info][upstream] external/envoy/source/common/upstream/cluster_manager_impl.cc:494] add/update cluster outbound|443||controller-manager-service.knative-sources.svc.cluster.local during init
> default/awssqs-devhose-source-vzrsc-765c547b68-gmp7m[istio-proxy]: [2018-11-20 17:42:53.580][19][info][upstream] external/envoy/source/common/upstream/cluster_manager_impl.cc:494] add/update cluster outbound|443||*.amazonaws.com during init

At this point, the receive adapter starts to poll successfully.

In my tests I had a plain exit if there was a poll error (I'm reading from a queue with a lot of messages so it was hard to read the logs while debugging.) I didn't realize that the sleep I added on error actually masks this behaviour.